python/res/enkf/jobs/scaling/measured_data.py

```diff
@@ -9,58 +9,79 @@
 
 
 class MeasuredData(object):
-    def __init__(self, ert, events):
-        self.data = self._get_data(ert, events.keys)
-        self.filter_on_column_index(events.index)
-        self.remove_nan(events.keys)
-        self.filter_out_outliers(events)
+    def __init__(self, ert, keys, index_lists=None):
+        if index_lists is None:
+            index_lists = [None] * len(keys)
+        self.data = self._get_data(ert, keys, index_lists)
 
-    def remove_nan(self, keys):
+    def remove_nan_and_filter(self, keys, std_cutoff, alpha):
+        self.remove_nan()
+        self.filter_out_outliers(keys, std_cutoff, alpha)
+
+    def remove_nan(self):
         """
         Removes NaN values from the case, first on a row basis to remove failed realizations,
         then on a column basis.
         """
-        self.data.loc[list(keys)] = self.data.loc[list(keys)].dropna(axis=0, how="all")
+        self.data[~self.data.index.isin(["OBS", "STD"])] = self.data[
+            ~self.data.index.isin(["OBS", "STD"])
+        ].dropna(axis=0, how="all")
         self.data = self.data.dropna(axis=1)
+        if self.data.shape[1] == 0:
+            raise ValueError("Empty dataset, all data has been filtered out")
 
-    def filter_on_column_index(self, index_list):
-        self.data = self._filter_on_column_index(index_list)
+    @staticmethod
+    def filter_on_column_index(dataframe, index_list):
+        return MeasuredData._filter_on_column_index(dataframe, index_list)
 
-    def filter_out_outliers(self, events):
-        self.data = self._filter_out_outliers(events)
+    def filter_out_outliers(self, keys, std_cutoff, alpha):
+        self.data = self._filter_out_outliers(keys, std_cutoff, alpha)
 
-    def _get_data(self, ert, observation_keys):
+    def _get_data(self, ert, observation_keys, index_lists):
         """
         Adds simulated and observed data and returns a dataframe where ensamble members will
-        have a data key, observed data will be prefixed with OBS_ and observated standard
-        deviation will be prefixed with STD_.
+        have a data key, observed data will be named OBS and observated standard
+        deviation will be named STD.
         """
         measured_data = pd.DataFrame()
         case_name = ert.getEnkfFsManager().getCurrentFileSystem().getCaseName()
-        for key in observation_keys:
+        for key, index_list in zip(observation_keys, index_lists):
```

Comment:
Maybe comment on why zip is used here?

Reply:
The change introduced here will allow the user to input an index list for each key, which wasn't possible before as a common was given for all keys. The calling function should make sure that an index list (could be None) exists for each key.

Reply:
:+1: